src/Microsoft.TemplateEngine.Cli/TemplateInvoker.cs

```diff
@@ -170,6 +164,27 @@ private async Task<CreationResultStatus> CreateTemplateAsync(ITemplateMatchInfo
                 return CreationResultStatus.CreateFailed;
             }
 
+            string fallbackName = new DirectoryInfo(
+                !string.IsNullOrWhiteSpace(_commandInput.OutputPath)
+                    ? _commandInput.OutputPath
+                    : Directory.GetCurrentDirectory())
+                .Name;
+
+            if (string.IsNullOrEmpty(fallbackName) || string.Equals(fallbackName, "/", StringComparison.Ordinal))
+            {   // DirectoryInfo("/").Name on *nix returns "/", as opposed to null or "".
+                fallbackName = null;
+            }
+            // Name returns <disk letter>:\ for root disk folder on Windows - replace invalid chars
+            else if (fallbackName.IndexOfAny(invalidChars) > -1)
+            {
+                Regex pattern = new Regex($"[{Regex.Escape(new string(invalidChars))}]");
```

Comment:
We don't need regex here. `string.Replace` already has an overload that accepts `char[]`.

Reply:
Sorry, I was thinking about `string.Split`. There is no such overload in `string.Replace`.